Дописал казино с учётом правок с код-ревью

alozhka opened the pull request.

alozhka committed:
Дописал казино с учётом правок с код-ревью

pklgn commented on Casino/Program.cs:
```diff
@@ -0,0 +1,45 @@
+{
+    public static void Main()
+    {
+        var rand = new Random();
```
Не используем `var`, лучше указывать явно тип данных.

pklgn commented on Casino/Program.cs:
```diff
@@ -0,0 +1,45 @@
+    {
+        var rand = new Random();
+        decimal balance = 10000;
+        const int multiplicator = 10;
```
Для констант используем PascalCase.
https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions

pklgn commented on Casino/Program.cs:
```diff
@@ -0,0 +1,45 @@
+            Console.WriteLine("\nБаланс: " + balance);
+            Console.Write("Введите ставку: ");
+            bool betParseResult = int.TryParse(Console.ReadLine(), out var bet);
+            if (betParseResult is false)
```
Лишняя проверка на `false`. Можно проще, назвав переменную в духе `isValidBet`. В этом случае читаемость условия `if (!isValidBet)` будет лучше.

pklgn commented on Casino/Program.cs:
```diff
@@ -0,0 +1,45 @@
+            else
+            {
+                int randomNumber = rand.Next(20) + 1;
+                if (randomNumber is 18 or 19 or 20)
```
Если значений для выигрыша будет больше, то такая проверка тяжело читается. Лучше вынести все выигрышные значения в отдельный массив и дальше проверять вхождения в него с помощью метода `Contains()`. Дока [тут](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.contains?view=net-8.0).

pklgn commented on Casino/Program.cs:
```diff
@@ -0,0 +1,45 @@
+                int randomNumber = rand.Next(20) + 1;
+                if (randomNumber is 18 or 19 or 20)
+                {
+                    var profit = bet * (1 + multiplicator * randomNumber % 17);
```
Не используем `var`.

pklgn commented on Casino/Program.cs:
```diff
@@ -0,0 +1,45 @@
+            }
+            else
+            {
+                int randomNumber = rand.Next(20) + 1;
```
Диапазон для случайного числа лучше вынести в константу.

pklgn replied:
Почему не вынес диапазон `20` в константу?

pklgn commented:
По логике работы приложения вопросов нет, все соответствует условию.

pklgn commented on Casino/Program.cs:
```diff
@@ -0,0 +1,45 @@
+                {
+                    var profit = bet * (1 + multiplicator * randomNumber % 17);
+                    balance += profit;
+                    Console.WriteLine("Выигрыш ставки: +" + profit);
```
Можно использовать интерполяцию строк. Дока [тут](https://learn.microsoft.com/en-us/dotnet/csharp/tutorials/string-interpolation).

alozhka committed:
Обновил приложение со всеми доработками

Вынес выйгрышные значения в константное поле, убрал var, сделал более понтные названия переменных

pklgn commented on Casino/Program.cs:
```diff
@@ -0,0 +1,46 @@
+        
+        while (balance > 0)
+        {
+            Console.WriteLine("\nБаланс: " + balance);
```
Здесь тоже интерполяция строк напрашивается.

pklgn commented:
Осталась пара мелких доработок: 
1. Интерполяция строк
2. Вынести диапазон 20 для случайных чисел в константу

alozhka committed:
Вынес максимальное случайное число в константу и сделал интерполяцию

pklgn reviewed (approved):
LGTM!
